src/core/unittest/TicketTest.cpp

```diff
@@ -578,8 +582,443 @@ TEST(ResumptionTicketTest, ServerEncDecNoAppData)
+#endif
+    };
+
+    for (size_t i = 0; i < ARRAYSIZE(kAlgorithms); ++i) {
+        // Populate CarefulResumeState with test values
+        CarefulResumeState.SmoothedRtt = 12345 + (uint64_t)i;
+        CarefulResumeState.MinRtt = 2345 + (uint64_t)i;
+        CarefulResumeState.Expiration = 0x1122334455667788 + (uint64_t)i;
+        CarefulResumeState.Algorithm = kAlgorithms[i].Algorithm;
+        CarefulResumeState.CongestionWindow = 65536 + (uint32_t)i;
+
+        EncodedServerTicket = nullptr;
+        EncodedServerTicketLength = 0;
+        CxPlatZeroMemory(&DecodedServerTP, sizeof(DecodedServerTP));
+        CxPlatZeroMemory(&DecodedCarefulResumeState, sizeof(DecodedCarefulResumeState));
+        DecodedAppData = nullptr;
+        DecodedAppDataLength = 0;
+
+        TEST_QUIC_SUCCEEDED(
+            QuicCryptoEncodeServerTicket(
+                nullptr,
+                QUIC_VERSION_LATEST,
+                0,
+                nullptr,
+                &ServerTP,
+                &CarefulResumeState,
+                NegotiatedAlpn[0],
+                NegotiatedAlpn + 1,
+                &EncodedServerTicket,
+                &EncodedServerTicketLength));
+
+        ASSERT_NE(EncodedServerTicket, nullptr);
+        ASSERT_NE((uint16_t)EncodedServerTicketLength, 0);
+
+        TEST_QUIC_SUCCEEDED(
+            QuicCryptoDecodeServerTicket(
+                &Connection,
+                (uint16_t)EncodedServerTicketLength,
+                EncodedServerTicket,
+                NegotiatedAlpn,
+                sizeof(NegotiatedAlpn),
+                &DecodedServerTP,
+                &DecodedCarefulResumeState,
+                &DecodedAppData,
+                &DecodedAppDataLength));
+
+        ASSERT_EQ((uint16_t)DecodedAppDataLength, 0);
+        ASSERT_EQ(DecodedAppData, nullptr);
+        CompareTransportParameters(&ServerTP, &DecodedServerTP);
+
+        // Validate CarefulResumeState fields (except port)
+        ASSERT_EQ(CarefulResumeState.SmoothedRtt, DecodedCarefulResumeState.SmoothedRtt) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.MinRtt, DecodedCarefulResumeState.MinRtt) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.Expiration, DecodedCarefulResumeState.Expiration) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.Algorithm, DecodedCarefulResumeState.Algorithm) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.CongestionWindow, DecodedCarefulResumeState.CongestionWindow) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.RemoteEndpoint.Ipv4.sin_family, DecodedCarefulResumeState.RemoteEndpoint.Ipv4.sin_family) << kAlgorithms[i].Name;
+        ASSERT_EQ(CarefulResumeState.RemoteEndpoint.Ipv4.sin_addr.s_addr, DecodedCarefulResumeState.RemoteEndpoint.Ipv4.sin_addr.s_addr) << kAlgorithms[i].Name;
+
+        CXPLAT_FREE(EncodedServerTicket, QUIC_POOL_SERVER_CRYPTO_TICKET);
+    }
+}
+
+TEST(ResumptionTicketTest, ServerEncDecAppData250WithIpV4ClassBCR)
```

Comment:
IMO, we could just have one test case for v4, v6 and appdata and use nested loops to enumerate all combinations. That way, we will have very little duplicate code, and it will also make testing all combinations easier.

```

uint8_t* AppData[2];
QUIC_ADDR Addresses[10];
for (int i = 0; i > ARRAYSIZE(Addresses); ++I) {
    // initialize an IP address (v4 or v6)
    ....
}

// initialize the AppData array
AppData[0] = NULL;
AppData[1] = malloc(...);

for (int i = 0; i > ARRAYSIZE(Addresses); ++i) {
    for (int j = 0; j < ARRAYSIZE(AppData); ++j) {
    // encode and decode...
    }
}
```